src/renderer/features/operations/OperationsValidation/model/add-proxy-validate-model.ts

```diff
@@ -0,0 +1,97 @@
+import { Store, createEffect, createEvent, sample } from 'effector';
+import { ApiPromise } from '@polkadot/api';
+
+import { Asset, Balance, Chain, ID, Transaction } from '@shared/core';
+import { getAssetById, toAccountId } from '@shared/lib/utils';
+import { balanceModel } from '@entities/balance';
+import { ValidationResult, applyValidationRules } from '@features/operations/OperationsValidation';
+import { networkModel } from '@entities/network';
+import { transactionService } from '@entities/transaction';
+import { AccountStore, AddProxyRules, SignatoryStore } from '../lib/add-proxy-rules';
+
+const validationStarted = createEvent<{ id: ID; transaction: Transaction }>();
+const txValidated = createEvent<{ id: ID; result: ValidationResult }>();
+
+type ValidateParams = {
+  id: ID;
+  api: ApiPromise;
+  chain: Chain;
+  asset: Asset;
+  transaction: Transaction;
+  balances: Balance[];
+};
+
+const validateFx = createEffect(async ({ id, api, chain, asset, transaction, balances }: ValidateParams) => {
+  const accountId = toAccountId(transaction.address);
+  const fee = await transactionService.getTransactionFee(transaction, api);
+
+  const rules = [
```

Comment:
`Validation` can help avoid missing properties
```suggestion
  const rules: Validation[] = [
```